### PR TITLE
synthetic data samples for questionnaire response

### PR DESCRIPTION
**Pull request by apeirats-va**

### CODE REVIEW

File: patient-generated-data-synthetic/src/test/java/gov/va/api/health/patientgenerateddata/Populaterator.java [hunk 1]
```diff
@@ -109,6 +113,7 @@ private static void populate(@NonNull Db db) {
   @SneakyThrows
   private static void questionnaire(@NonNull Connection connection) {
     Set<String> ids = new HashSet<>();
+    String sql = sql("app.questionnaire", List.of("id", "payload", "version"));
```
Comment (jblefkove-va) [inline patient-generated-data-synthetic/src/test/java/gov/va/api/health/patientgenerateddata/Populaterator.java:116]: Define this closer to where it is used

File: patient-generated-data-synthetic/src/test/java/gov/va/api/health/patientgenerateddata/Populaterator.java [hunk 2]
```diff
@@ -137,6 +134,43 @@ private static void questionnaire(@NonNull Connection connection) {
...
+        statement.setObject(1, id);
+        statement.setObject(2, new ObjectMapper().writeValueAsString(response));
+        statement.setObject(3, 0);
+        statement.execute();
+      }
+    }
+  }
+
+  private static String sql(String table, Collection<String> columns) {
+    Objects.requireNonNull(table);
```
Comment (jblefkove-va) [inline patient-generated-data-synthetic/src/test/java/gov/va/api/health/patientgenerateddata/Populaterator.java:163]: You can put the lombok `@NonNull` check directly on method parameters for this

File: patient-generated-data-synthetic/src/test/java/gov/va/api/health/patientgenerateddata/Populaterator.java [hunk 3]
```diff
@@ -137,6 +134,43 @@ private static void questionnaire(@NonNull Connection connection) {
...
+      try (PreparedStatement statement = connection.prepareStatement(sql)) {
+        statement.setObject(1, id);
+        statement.setObject(2, new ObjectMapper().writeValueAsString(response));
+        statement.setObject(3, 0);
+        statement.execute();
+      }
+    }
+  }
+
+  private static String sql(String table, Collection<String> columns) {
```
Comment (jblefkove-va) [inline patient-generated-data-synthetic/src/test/java/gov/va/api/health/patientgenerateddata/Populaterator.java:162]: `sqlInsert`?

**Comment by jblefkove-va** [comment]
github is flagging no newline at end-of-file for the two json documents

File: patient-generated-data-synthetic/src/test/java/gov/va/api/health/patientgenerateddata/Populaterator.java (hunk 3 above)
Comment (apeirats-va) [reply patient-generated-data-synthetic/src/test/java/gov/va/api/health/patientgenerateddata/Populaterator.java:162]: sure

File: patient-generated-data/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -33,3 +33,42 @@ databaseChangeLog:
...
+                  type: clob
+                  constraints:
+                    nullable: false
+              - column:
+                  name: version
+                  type: int
+                  constraints:
+                    nullable: false
+
+  - changeSet:
```
Comment (jblefkove-va) [inline patient-generated-data/src/main/resources/db/changelog/db.changelog-master.yaml:62]: nitpick, but the table rename should go before the new table

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**